analyses/hotspots-detection/01-reccurence-hotspot-overlap.Rmd

```diff
@@ -0,0 +1,153 @@
+internal_hotspot_file <- params$internal_hotspot_file
+
+results_dir <- file.path(root_dir,
+                         "analyses",
+                         "hotspots-detection",
+                         "results")
+
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+```
+
+## Cancer hotspot files
+```{r}
+
+# Cancer database
+hotspot_database_2017_snv <- readxl::read_xls(mskcc_cancer_hotspot_file,sheet = 1) 
+hotspot_database_2017_indel <- readxl::read_xls(mskcc_cancer_hotspot_file,sheet = 2)
+
+hotspot_database <- bind_rows( hotspot_database_2017_snv, hotspot_database_2017_indel) %>%
+  dplyr::select("Amino_Acid_Position","Hugo_Symbol") %>%
+  dplyr::mutate(hotspot_database="MSKCC") %>%
+  unique()
+
+```
+
+## Gene table for reccurence checks
+```{r}
+gene_table <- read_tsv("../fusion_filtering/references/genelistreference.txt")%>%
+  dplyr::rename(Hugo_Symbol=Gene_Symbol) %>%
+  dplyr::select(Hugo_Symbol,type)
+
+brain_goi <- read_tsv("input/brain-goi-list-new.txt",col_names = "Hugo_Symbol") %>%
+	dplyr::mutate(type="brain-goi")
+
+gene_table <- bind_rows(gene_table,
+			brain_goi)
+
+```
+
+## Read database 
+```{r}
+
+db_file <- file.path(root_dir, params$db_file)
+
+# Start up connection
+con <- DBI::dbConnect(RSQLite::SQLite(), db_file,path = ":memory:")
+
+```
+
+## Designate caller tables from SQL file
+```{r}
+strelka <- dplyr::tbl(con, "strelka")
+mutect <- dplyr::tbl(con, "mutect")
+vardict <- dplyr::tbl(con, "vardict")
+lancet <- dplyr::tbl(con, "lancet") 	
+
+```
+
+## Subset maf per caller
+```{r}
+source ("utils/prepMaf.R")
+strelka_subset<-prepMaf(strelka, gene_table = gene_table) %>%
+       mutate(caller="strelka")	
+mutect_subset<-prepMaf(mutect, gene_table = gene_table) %>%
+	mutate(caller="mutect")
+vardict_subset<-prepMaf(vardict, gene_table = gene_table) %>%
+	mutate(caller="vardict")
+lancet_subset<-prepMaf(lancet, gene_table = gene_table) %>%
+	mutate(caller="lancet")
+
+# combine
+combined_maf_subsets <-bind_rows(strelka_subset,
+mutect_subset,
+vardict_subset,
+lancet_subset)
+
+# save for future runs
+saveRDS(combined_maf_subsets, "results/combined_maf_subsets.RDS")
+
+	      
+```
+
+## Recurrence in genes part of brain-goi from filtered snv calls
+```{r}
+
+calls_recurrence <- combined_maf_subsets %>%
+    left_join(gene_table,by="Hugo_Symbol") %>%
+  group_by(Chromosome,Start_Position,End_Position,
+           Amino_Acid_Position,Hugo_Symbol,
+           Tumor_Sample_Barcode) %>%
+  summarise(type=toString(unique(type))) %>%
+    count(Chromosome,Start_Position,End_Position,Amino_Acid_Position,Hugo_Symbol,type,sort = TRUE) %>%
+  dplyr::filter( grepl("brain-goi",type)) %>%
+  dplyr::filter(n>2)
+
+
+```
+
+
+## Merge hotspot annotation to recurrence table and save fot strelka
+```{r}
+calls_recurrence <- calls_recurrence %>%
+	left_join(hotspot_database,by=c("Amino_Acid_Position","Hugo_Symbol"))%>%
+	write_tsv(file.path(results_dir,"snv_recurrence.tsv"))
```

Comment:
Hi @kgaonkar6! Nice work on this. I have a few more suggestions.

After this step, I think it would be good to add some summary information. For ex, when I looked at the mutations not in the MSK list, I see 270 and 61 genes. It would also be good to look at distribution of VAFs.

For genes that have multiple mutations, I think we should plot some lollipop diagrams.

Another QC step we should do is make sure that none of these are SNPs.

One thing I noticed when inspecting AKT1 for ex, was that many amino acid positions are NA and one is `-`. In looking at [pedcbio](https://pedcbioportal.kidsfirstdrc.org/results/mutations?cancer_study_list=pbta_all&tab_index=tab_visualize&case_set_id=pbta_all_all&Action=Submit&gene_list=AKT1), it looks like many of these are splice variants. All of them are at low VAF as well, so maybe we should discuss with David on our call tomorrow.